src/consensus/params.cpp

```diff
@@ -8,12 +8,30 @@
 #include <key_io.h>
 #include <script/standard.h>
 #include "upgrades.h"
+#include "util.h"
 
 namespace Consensus {
     bool Params::NetworkUpgradeActive(int nHeight, Consensus::UpgradeIndex idx) const {
         return NetworkUpgradeState(nHeight, *this, idx) == UPGRADE_ACTIVE;
     }
 
+    bool Params::FeatureActive(const int nHeight, const Consensus::ConsensusFeature feature) const {
+        auto requires = WhatDependsOn(feature);
+        if (vRequiredFeatures.count(feature) > 0 ||
+            std::any_of(requires.begin(), requires.end(), [&](Consensus::ConsensusFeature feat) { return FeatureActive(nHeight, feat); })) {
+            // Either the feature is required, or it is a dependency of an active feature.
```

Comment:
Scenario: we enable a feature at a network upgrade, not realizing we're enabling another feature at the same network upgrade. The `switch` statement below is not consulted and could have mismatching logic -- such as indicating that the feature is disabled, enabled in the wrong upgrade, etc.

We should assert that the result of the switch statement below is to `return true` if an active/required feature depends on our feature.

Reply:
This took me a moment, but I think it makes sense to me - we want to check the feature dependency path to ensure that none of the features are explicitly disabled, is that your meaning?

Reply:
Fixed in https://github.com/zcash/zcash/pull/4480/commits/d07294a851a7d1408cb01de531e33c41577ddc2e